controllers/setup.go

```diff
@@ -139,13 +146,17 @@ func setupManager(ctx context.Context, cancel context.CancelFunc, mgr controller
 		return fmt.Errorf("error adding service controller: %w", err)
 	}
 
-	vmController, err := CreateVmController(mgr)
-	if err != nil {
-		return fmt.Errorf("failed to create vm controller: %w", err)
-	}
+	if crdWatch.CrdExists(vmCRD) {
+		vmController, cErr := CreateVmController(mgr)
```

Comment:
nit, just for info: It is ok to use `err` variable name instead of `cErr`. It would shadow the `err` the outer context, but that does not matter, because if `cErr != nil` we always return the error.

You don't have to fix this.